### PR TITLE
FXIOS-1401 ⁃ Fixes: #7844 - Create equivalent XCUI smoke tests Xcode Test Plan

### PR DESCRIPTION
**Pull request by AaronMT**
See #7844 

> I can take https://github.com/mozilla-mobile/mobile-test-eng/issues/165
> 
> Going to split this up as I'm nervous about points 3 and 4 and any workflows that run the existing schema.
> 
> - Create equivalent Xcode Test Plan with the same enabled tests as Fennec_Enterprise_SmokeXCUITests
> - Remove Fennec_Enterprise_SmokeXCUITests schema
> - Reflect changes in Bitrise YML
> - Anything else? (@isabelrios)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1401)

### CODE REVIEW

**Comment by isabelrios** [comment]
This is looking good! thanks for picking this @AaronMT. 
I'm testing this locally and looks like it is fine. There are a few things pending after a first review:
- Remove the SmokeTest schema
- Add the option to the xcodebuild option in the yml file (or in Bitrise) `-testPlan SmokeXCUITests.xctestplan` to the xcode build and xcode test steps

I will spend some more time on this in case there are more comments.
Thanks

**Comment by AaronMT** [comment]
@isabelrios Thanks for the initial look.

I am looking at the ```bitrise.yml``` file recently added to ```main``` and am a little uncertain on one reference to the scheme ```Fennec_Enterprise_SmokeXCUITests``` as ```TestingWorkflowAllTestsScenarios``` currently looks for it, e.g,  

https://github.com/mozilla-mobile/firefox-ios/blob/b04d3ed46450075e10804a4e56778ec0a32ff3a6/bitrise.yml#L47

Any thoughts on how we might update these? 

---

And it looks the line we want to update is here 

https://github.com/mozilla-mobile/firefox-ios/blob/b04d3ed46450075e10804a4e56778ec0a32ff3a6/bitrise.yml#L1135

and add ```- xcodebuild_test_options: '-testPlan SmokeXCUITests'

**Comment by isabelrios** [comment]
> I am looking at the `bitrise.yml` file recently added to `main` and am a little uncertain on one reference to the scheme `Fennec_Enterprise_SmokeXCUITests` as `TestingWorkflowAllTestsScenarios` currently looks for it, e.g,
> 
> https://github.com/mozilla-mobile/firefox-ios/blob/b04d3ed46450075e10804a4e56778ec0a32ff3a6/bitrise.yml#L47
> 
> Any thoughts on how we might update these?

We will need to remove the references to that scheme name and use Fennec_Enterprise_XCUITests instead adding the option that you mentioned below in the xcodebuild_test_options so that the proper tests are run

> and replace with `- xcodebuild_test_options: '-testPlan SmokeXCUITests`

**Comment by isabelrios** [comment]
I think that would work, but we will need to test carefully to be 100% sure about that. I may be missing something else that could break the workflows

**Comment by AaronMT** [comment]
Thanks. Let's try it I suppose. I've added a commit, and we can run it (@isabelrios).

File: bitrise.yml [hunk 1]
```diff
@@ -1132,7 +1132,7 @@ workflows:
         is_always_run: true
     - xcode-test@2:
         inputs:
-        - scheme: Fennec_Enterprise_SmokeXCUITests
```
Comment (isabelrios) [inline bitrise.yml:1135]: I think we still need to have a scheme here, now: Fennec_Enterprise_XCUITest.

File: bitrise.yml [hunk 2]
```diff
@@ -1880,7 +1880,7 @@ workflows:
         - xcodebuild_options: >-
             CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO
             CODE_SIGNING_ALLOWED=NO
-        - scheme: Fennec_Enterprise_SmokeXCUITests
```
Comment (isabelrios) [inline bitrise.yml:1883]: Same as above

File: bitrise.yml (hunk 1 above)
Comment (AaronMT) [reply bitrise.yml:1135]: Ah they still need a scheme that's right. Thanks.

**Pull request merged**